lib/trino-filesystem-azure/src/main/java/io/trino/filesystem/azure/AzureFileSystem.java

```diff
@@ -370,6 +371,33 @@ public Set<Location> listDirectories(Location location)
         }
     }
 
+    @Override
+    public Optional<Location> createTemporaryDirectory(Location targetPath, String temporaryPrefix, String relativePrefix)
+            throws IOException
+    {
+        AzureLocation azureLocation = new AzureLocation(targetPath);
+        if (!isHierarchicalNamespaceEnabled(azureLocation)) {
+            return Optional.empty();
+        }
+
+        Location temporary;
+        if (temporaryPrefix.startsWith("/")) {
```

Comment:
add a comment on why this is needed... specifically I don't see the same treatment in HDFS

Reply:
This is because we do `new Path(target, prefix)` for HDFS which will replace the path when `prefix` is absolute. We have to implement that manually for Azure. I'll add a comment explaining what it is doing.